File: 347-top-k-frequent-elements/top-k-frequent-elements.js
/**
 * @param {number[]} nums
 * @param {number} k
 * @return {number[]}
 */
var topKFrequent = function (nums, k) {
    // map an object with number and count
    // sort it by frequency
    const numsMap = {};
    for (let i = 0; i < nums.length; i++) {
        const num = nums[i];
        if (numsMap[num] === undefined) numsMap[num] = 0;
        numsMap[num] += 1;
    }
    var array = Object.keys(numsMap).map((key) => [key, numsMap[key]]);
    array.sort((a, b) => a[1] - b[1]);
    const result = [];
    for(let i = 0; i<k; i++){
        result.push(parseInt(array.pop()[0]));
    }
    return result;
};

File: 347-top-k-frequent-elements/top-k-frequent-elements.rb
# @param {Integer[]} nums
# @param {Integer} k
# @return {Integer[]}
def top_k_frequent(nums, k)
    nums_hash = {}
    nums.each do |num|
        nums_hash[num] = 0 if nums_hash[num] == nil
        nums_hash[num] +=1
    end
    nums_array = nums_hash.sort_by{|key, value| value}.to_h.keys.last(k)
    return nums_array
end